### DISCUSSION TITLE
Integration breaks on HA reload "Failed Setup, will retry" - argument of type 'noneType' is not iterable

### DISCUSSION THREAD

**Discussion by flewis88** [petretiandrea/home-assistant-tapo-p100#688] [Q&A]
On reload of HA, integration of Tapo devices breaks and become uncontactable despite working in the native Tapo app. Devices are shown as "Failed Setup, will retry", and mouse over the error reads "argument of type 'noneType' is not iterable". reload or retry doesn't fix it,but if devices are deleted and re-added, they work fine. Not sure if I've missed part of the process. I have the dynamic DNS tracking turned on.

**Comment by ryancodingfun** [comment]
I am also getting the same issue with the L530 light bulbs. I wish it could retain the username/password.

**Reply by flewis88** [reply]
I was able to resolve it by turning off the dynamic MAC or IP setting that it offers.
